### ISSUE TITLE
Simplify interface of PJ_COORD etc. and make immutable

### ISSUE THREAD

**Issue by johannessen** [johannessen/proj-perl-ffi#2]
The current approach of trying to “emulate” PROJ’s `union` of `struct`s (in `PJ_COORD`) as exactly as possible with FFI in Perl doesn’t appear to be very useful. The many method calls cause a performance drain. Additionally, the implementation is difficult to maintain.

The complex named parameter constructors for `PJ_COORD`, `PJ_UV` etc. should be removed in favour of calling `proj_coord()`.

`PJ_COORD` and friends should be implemented in a simplified fashion for Perl:
```perl
# Instead of this:
$z = $coord->xyz->z;

# The interface should look like this:
$z = $coord->xyz_z;
```

Such an interface wouldn’t preclude adding mutability later on, but in practice, I believe mutable coordinates are almost never needed, so ultimately this may not happen.

**Comment by johannessen** [comment]
Given the explicit warning under ["DATA TYPES" in Geo::LibProj::FFI](https://metacpan.org/release/AJNN/Geo-LibProj-FFI-0.04/view/lib/Geo/LibProj/FFI.pm#DATA-TYPES), breaking backwards compatibility with this change would appear to be acceptable.

**Comment by johannessen** [comment]
Resolved with [0.05](https://metacpan.org/release/AJNN/Geo-LibProj-FFI-0.05).
